test/1_stdlib/Print.swift

```diff
@@ -169,7 +169,7 @@ func test_StdlibTypesPrinted() {
     var optionalString: String? = nil
     printedIs(optionalString, "nil")
     optionalString = "meow"
-    printedIs(optionalString, "Optional(\"meow\")")
+    printedIs(optionalString, "meow")
```

Comment:
Sure, this is the behavior change that I wanted reviewed.

This happens because _print_unlocked first checks whether the value is convertible to CustomDebugStringConvertible. String is. With my change we correctly recognize that and just print the string itself. Previously we would fall into ad-hoc printing that reflects the type.

I could modify stdlib to special case printing of optionals, but I don't know why we would want to preserve the old behavior.